### PR TITLE
fix: handle stdin end

### PR DESCRIPTION
**Pull request by rishabh3112**
**What kind of change does this PR introduce?**
Ends main CLI process when stdin ends.

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
No

**Summary**
Attempt at fixing stdin issue

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

File: packages/webpack-cli/bin/cli.js [hunk 1]
```diff
@@ -18,6 +18,18 @@ if (importLocal(__filename)) {
 
 process.title = 'webpack';
 
+// Exit process when no data is to be consumed
+process.stdin.on('end', () => {
+    process.exit(0);
+});
+
+// Exit process when stream closes
+process.stdin.on('close', () => {
```
Comment (alexander-akait) [inline packages/webpack-cli/bin/cli.js:27]: Why on `close`? I think `end` is enough

Comment (rishabh3112) [reply packages/webpack-cli/bin/cli.js:27]: Yup, changing.

**Comment by mveytsman** [comment]
Hello!

I think this issue is blocking upgrading webpack in the Phoenix framework (see https://github.com/phoenixframework/phoenix/pull/4054). Is there a timeline on getting this PR in? 

Also, just curious, it looks like this will supersede `--watch-options-stdin` and webpack will always exit when stdin is closed. Is that correct?

**Comment by alexander-akait** [comment]
@mveytsman I think we will fix it today/tomorrow and do release, we should respect `--watch-options-stdin` option, you are right

**Pull request closed without merging**